File: 49-group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        m = {}
        for s in strs:
            ss = ''.join(sorted(s))
            if ss in m:
                m[ss].append(s)
            else:
                m[ss] = [s]
        return m.values()

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> m = new HashMap<>();
        for (String s: strs) {
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String sortedS = new String(charArray);
            m.putIfAbsent(sortedS, new ArrayList<>());
            m.get(sortedS).add(s);
        }
        return new ArrayList<>(m.values());
    }
}